### PR TITLE
feat: worker deploy and delete

### LINKED ISSUE #36: Preview deployments for Workers
This is not specific plugin but a nice-to-have features for plugin Workers.

Currently when developing it's quite easy to switch to a different Action branch for testing a new feature, by specifying the `@branch` within the configuration.

However, workers do not benefit from this and always have to be deployed from the main branch to be tested. This makes development more tedious than it has to be, even worse when testing other people pull-requests.

What would be nice is to get a preview URL base on the latest commit of the branch the user is working on.

It would also be important to cleanup the preview deployments as they would pollute Cloudflare heavily.

A naming pattern could be:
`<plugin-name>.<username>.<branch>.<hash>` to find them easily and know what they refer to. We could also have the deployed url displayed in the Action results to be found easily, and without the need to access Github token because we do not publish in the pull-request thread.

### PR DESCRIPTION
**Pull request by gentlementlegen**
Resolves https://github.com/ubiquibot/plugins-wishlist/issues/36

## QA
Deploying: https://github.com/Meniole/command-start-stop/actions/runs/10590231101
Deleting: https://github.com/Meniole/command-start-stop/actions/runs/10590234079

### CODE REVIEW

**Comment by 0x4007** [comment]
We can only have [100 workers](https://developers.cloudflare.com/workers/platform/limits/#number-of-workers) on our account. This seems like a bad solution. This should deploy on each developer's personal cloudflare accounts instead, [like I mentioned](https://github.com/ubiquibot/plugins-wishlist/issues/36#issuecomment-2303382955) in the issue discussion. 

It could be interesting to register a cloudflare key in a similar way we do wallet registrations per user. We can allow them to encrypt using our X25519 key with an adjustment to that evm private key encryption app. 

Then they can safely post their encrypted cloudflare key to github comments and nobody but the bot can use it.

**Comment by gentlementlegen** [comment]
It deploys on the owners account meaning pull-requests won't deploy on our Cloudflare, but inside the author of the PR, so we should be fine and the user can provide directly the URL of the deployed worker for testing. It is very easy to fork to test on our personal Cloudflare, so we should be fine, except if we have 20 branches in each of our repos.

**Comment by 0x4007** [comment]
> It deploys on the owners account meaning pull-requests won't deploy on our Cloudflare, but inside the author of the PR, so we should be fine and the user can provide directly the URL of the deployed worker for testing. 

How does it deploy on the contributors account? I just wrote a new task for that.

**Comment by gentlementlegen** [comment]
@0x4007 It deploys on branch created / pushed to. When a user opens a pull-request, this does not create a branch on our own repository but on the contributor's one, meaning the workflow will be triggered on its own repo but not our (so basically deploying to the user's repo). So on our side we will always just have `development` and `main` deployments, plus any custom branch we create within the Organization.

**Comment by 0x4007** [comment]
I don't understand how it could possibly post on another Cloudflare account without authentication

**Comment by gentlementlegen** [comment]
The contributor has to setup its own Cloudflare for this to work, same as within your ticket https://github.com/ubiquibot/plugin-template/issues/17

**Comment by 0x4007** [comment]
The answer is that the contributor sets the key in their fork's secrets, and then the CI will run from their fork, using their token.

**Comment by 0x4007** [comment]
> ## QA
> Deploying: https://github.com/Meniole/command-start-stop/actions/runs/10590231101 

That is a crazy branch name.

**Review by 0x4007 — APPROVED** [review]
Looks like it works according to your QA

**Comment by gentlementlegen** [comment]
> > ## QA
> > Deploying: https://github.com/Meniole/command-start-stop/actions/runs/10590231101
> 
> That is a crazy branch name.

Hopefully no one names it this way, but we surely can have plenty of levels when forks are tested like `/fork/ubq/fix/some-bug-fix`. Special characters was just to make sure `echo` commands don't break and urls make sense.

**Comment by gentlementlegen** [comment]
Had to make modifications because of [RFC 1035](https://datatracker.ietf.org/doc/html/rfc1035#section-2.3.4) allowing only 63 characters.

File: .github/workflows/worker-delete.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+      - name: Get Deleted Branch Name
+        id: get_branch
+        run: |
+          branch_name=$(echo '${{ github.event.ref }}' | sed 's#refs/heads/##' | sed 's#[^a-zA-Z0-9]#-#g')
+          echo "branch_name=$branch_name" >> $GITHUB_ENV
+      - name: Retrieve and Construct Full Worker Name
+        id: construct_worker_name
+        run: |
+          base_name=$(grep '^name = ' wrangler.toml | sed 's/^name = "\(.*\)"$/\1/')
```
Comment (whilefoo) [inline .github/workflows/worker-delete.yml:29]: won't this grep match multiple lines for example [a sample config](https://developers.cloudflare.com/workers/wrangler/configuration/#sample-wranglertoml-configuration) that has two lines that start with `name = `

**Review by whilefoo — APPROVED** [review]
that's super useful!

File: .github/workflows/worker-delete.yml (hunk 1 above)
Comment (gentlementlegen) [reply .github/workflows/worker-delete.yml:29]: Good point, I didn't think of this scenario. I guess we should always base it on the top level `name` to generate the `base_name`. To be sure it's the top level I guess I should read the `.toml` and store the values in an object, which complexifies this script quite a lot. Mind if I take care of it in another pull-request? I think this is quite helpful to get in first for the dev experience and we did not encounter this scenario yet.

**Comment by gentlementlegen** [comment]
Opened https://github.com/ubiquibot/plugin-template/issues/18 to take care of the multiple `name` issue, will probably have to manually open a bunch of PRs on our Worker Plugins to get these changes in.

**Pull request merged**